pinc/Project.inc

```diff
@@ -566,6 +566,23 @@ XML;
 
         return $updated_record;
     }
+
+    // check if project has entered a formatting round
+    public function has_entered_formatting_round()
+    {
+        global $PROJECT_STATES_IN_ORDER;
+        $state_index = array_search($this->state, $PROJECT_STATES_IN_ORDER);
+        for($i=$state_index; $i>=0; $i--)
+        {
+            $round = get_Round_for_project_state($PROJECT_STATES_IN_ORDER[$i]);
+            if($round)
```

Comment:
You could simplify this somewhat by:
```
if($round && is_formatting_round($round))
```
Totally optional.

Reply:
Although you need to include_once `stages.inc` at the top of this file if you're using a function from it.